v2/yarpchttpfx/yarpchttpfx.go

```diff
@@ -157,6 +158,7 @@ type ClientParams struct {
 	Config          OutboundsConfig
 	DialerProvider  yarpc.DialerProvider
 	ChooserProvider yarpc.ChooserProvider
+	Middleware      []yarpc.UnaryOutboundTransportMiddleware `name:"yarpcfx"`
```

Comment:
I really like this idea actually. It eliminates the need to duplicate similar middleware processing in each of the transports and consolidates it all in one place. I'll make this change here!

Reply:
Although, this would mean that we mutate the `yarpc.Client` internals from within the `yarpcfx` module. It would look something along the lines of,
```
// NewClientProvider provides a yarpc.ClientProvider to the Fx application.
func NewClientProvider(p ClientProviderParams) (ClientProviderResult, error) {
	clients := p.Clients
	for _, cl := range p.ClientLists {
		clients = append(clients, cl...)
	}
	for _, c := range clients {
		c.Unary = yarpc.ApplyUnaryOutboundTransportMiddleware(c.Unary, p.UnaryOutboundTransportMiddleware...)
	}
	provider, err := yarpcclient.NewProvider(clients...)
	if err != nil {
		return ClientProviderResult{}, err
	}
	return ClientProviderResult{
		Provider: provider,
	}, nil
}
```

This seems fine if we feel strongly about consolidating the application of the middleware, but it's a point worth considering.

@AlexAPB @kriskowal Thoughts?

Reply:
🤔 I think I'm okay with it since we pass clients by value, but I admit that it's not completely ergonomic

Reply:
True, but we can still mutate the internal `Unary` outbound. I made the changes so that we can more easily compare and contrast the approaches.

Reply:
`ApplyMiddlewareToClients(Clients)Clients`

Reply:
I'd prefer to leave this as is for now since if we introduce and apply transport middleware for streams here, the function signature would be fairly noisy:
```
applyMiddlweareToClients(clients []yarpc.Client, unaryMiddleware []yarpc.UnaryOutboundTransportMiddleware, streamMiddleware []yarpc.StreamOutboundTransportMiddlware) []yarpc.Client { ... }
```

I'm happy to revisit this if we feel strongly about it.